tests/scripts/check-names.sh

```diff
@@ -16,12 +16,22 @@ if grep --version|head -n1|grep GNU >/dev/null; then :; else
     exit 1
 fi
 
+trace=
+if [ $# -ne 0 ] && [ "$1" = "-v" ]; then
+  shift
+  trace='-x'
+  exec 2>check-names.err
+  trap 'echo "FAILED UNEXPECTEDLY, status=$?";
+        cat check-names.err' EXIT
+  set -x
+fi
+
 printf "Analysing source code...\n"
 
-tests/scripts/list-macros.sh
+sh $trace tests/scripts/list-macros.sh
 tests/scripts/list-enum-consts.pl
```

Comment:
If the error ends up being from list-enum-consts.pl, how can we know? Just that it's the only one not traced?

Reply:
We'll know because the last line of the trace is `list-enum-consts.pl`. We won't know why it's crashed. Perl doesn't have a dead-simple trace functionality like sh, but hopefully if it was failing it would print an exception backtrace.